php 7.3 

diff --git a/pkgs/development/interpreters/php/default.nix b/pkgs/development/interpreters/php/default.nix
index b97c37b0cf5..1a6b90ba82e 100644
--- a/pkgs/development/interpreters/php/default.nix
+++ b/pkgs/development/interpreters/php/default.nix
@@ -2,9 +2,11 @@
 
 { lib, stdenv, fetchurl, composableDerivation, autoconf, automake, flex, bison
 , mysql, libxml2, readline, zlib, curl, postgresql, gettext
-, openssl, pcre, pkgconfig, sqlite, config, libjpeg, libpng, freetype
+, openssl, pcre, pcre2, pkgconfig, sqlite, config, libjpeg, libpng, freetype
 , libxslt, libmcrypt, bzip2, icu, openldap, cyrus_sasl, libmhash, freetds
-, uwimap, pam, gmp, apacheHttpd, libiconv, systemd, libsodium }:
+, uwimap, pam, gmp, apacheHttpd, libiconv, systemd, libsodium, libargon2
+, libzip
+}:
 
 let
 
@@ -230,6 +232,15 @@ let
           configureFlags = ["--with-sodium=${libsodium.dev}"];
           buildInputs = [libsodium];
         };
+
+	argon2 = {
+	  configureFlags = ["--with-password-argon2=${libargon2}"];
+	};
+
+	libzip = {
+	  configureFlags = ["--with-libzip=${libzip.dev}"];
+	  buildInputs = [libzip];
+	};
       };
 
       cfg = {
@@ -270,6 +281,8 @@ let
         ztsSupport = config.php.zts or false;
         calendarSupport = config.php.calendar or true;
         sodiumSupport = (lib.versionAtLeast version "7.2") && config.php.sodium or true;
+        argon2Support = (config.php.argon2 or true) && (lib.versionAtLeast version "7.2");
+        libzipSupport = (config.php.libzip or true) && (lib.versionAtLeast version "7.2");
       };
 
       hardeningDisable = [ "bindnow" ];
@@ -294,8 +307,9 @@ let
 
       configureFlags = [
         "--with-config-file-scan-dir=/etc/php.d"
-        "--with-pcre-regex=${pcre.dev} PCRE_LIBDIR=${pcre}"
-      ] ++ lib.optional stdenv.isDarwin "--with-iconv=${libiconv}"
+      ] ++ lib.optional (lib.versionOlder version "7.3") "--with-pcre-regex=${pcre.dev} PCRE_LIBDIR=${pcre}"
+        ++ lib.optional (lib.versionAtLeast version "7.3") "--with-pcre-regex=${pcre2.dev} PCRE_LIBDIR=${pcre2}"
+        ++ lib.optional stdenv.isDarwin "--with-iconv=${libiconv}"
         ++ lib.optional stdenv.isLinux  "--with-fpm-systemd";
 
       postInstall = ''
@@ -356,4 +370,9 @@ in {
     version = "7.2.8";
     sha256 = "1rky321gcvjm0npbfd4bznh36an0y14viqcvn4yzy3x643sni00z";
   };
+
+  php73 = generic {
+    version = "7.3.3";
+    sha256 = "1riw0a1mzc5ymaj02rni57l5pyfkxl0ygf1l39q7ksnz7aa9x5k1";
+  };
 }
diff --git a/pkgs/development/libraries/pcre2/10_32.nix b/pkgs/development/libraries/pcre2/10_32.nix
new file mode 100644
index 00000000000..06cf0130f0d
--- /dev/null
+++ b/pkgs/development/libraries/pcre2/10_32.nix
@@ -0,0 +1,30 @@
+{ stdenv, fetchurl, fetchpatch }:
+
+stdenv.mkDerivation rec {
+  name = "pcre2-${version}";
+  version = "10.32";
+  src = fetchurl {
+    url = "ftp://ftp.csx.cam.ac.uk/pub/software/programming/pcre/${name}.tar.bz2";
+    sha256 = "0bkwp2czcckvvbdls7b331cad11rxsm020aqhrbz84z8bp68k7pj";
+  };
+
+  configureFlags = [
+    "--enable-pcre2-16"
+    "--enable-pcre2-32"
+    "--enable-jit"
+  ];
+
+  outputs = [ "bin" "dev" "out" "doc" "man" "devdoc" ];
+
+  postFixup = ''
+    moveToOutput bin/pcre2-config "$dev"
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Perl Compatible Regular Expressions";
+    homepage = http://www.pcre.org/;
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ ttuegel ];
+    platforms = platforms.all;
+  };
+}
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index dc1ddf2f84d..376532f61dc 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7141,6 +7141,10 @@ with pkgs;
     php = php72;
   });
 
+  php73Packages = recurseIntoAttrs (callPackage ./php-packages.nix {
+    php = php73;
+  });
+
   php_fpm53 = callPackage ../development/interpreters/php/5.3.nix {
     config = config // {
       php = (config.php or {}) // {
@@ -7159,6 +7163,11 @@ with pkgs;
     php70
     php71
     php72;
+    #php73;
+
+  php73 = callPackage ../development/interpreters/php {
+    pcre2 = pcre2_10_32;
+  };
 
   php-embed = php72-embed;
 
@@ -10935,6 +10944,7 @@ with pkgs;
   pcre-cpp = self.pcre.override { variant = "cpp"; };
 
   pcre2 = callPackage ../development/libraries/pcre2 { };
+  pcre2_10_32 = callPackage ../development/libraries/pcre2/10_32.nix { };
 
   pdf2xml = callPackage ../development/libraries/pdf2xml {} ;
 
