php53-fpm service

diff --git a/nixos/modules/module-list.nix b/nixos/modules/module-list.nix
index 0617279..2c98476 100644
--- a/nixos/modules/module-list.nix
+++ b/nixos/modules/module-list.nix
@@ -488,6 +488,7 @@
   ./services/web-servers/lighttpd/gitweb.nix
   ./services/web-servers/lighttpd/inginious.nix
   ./services/web-servers/nginx/default.nix
+  ./services/web-servers/php53-fpm.nix
   ./services/web-servers/phpfpm/default.nix
   ./services/web-servers/shellinabox.nix
   ./services/web-servers/tomcat.nix
diff --git a/nixos/modules/services/web-servers/php53-fpm.nix b/nixos/modules/services/web-servers/php53-fpm.nix
new file mode 100644
index 0000000..f6faf51
--- /dev/null
+++ b/nixos/modules/services/web-servers/php53-fpm.nix
@@ -0,0 +1,76 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.php53-fpm;
+  pkg = pkgs.php_fpm53;
+  valOpt = p : name : if ! isNull p then " --${name} ${toString p}" else "";
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.php53-fpm = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable php53-fpm service.";
+      };
+
+      #  -p, --prefix <dir>
+      prefix = mkOption {
+        default = null;
+        type = types.nullOr types.path;
+        example = "/";
+        description = ''
+          Specify alternative prefix path to FastCGI process manager. Default
+          is php-fpm's /nix/store/...-php-xxx/ installation directory.
+        '';
+      };
+
+      # -y, --fpm-config <file>
+      configFile = mkOption {
+        default = null;
+        type = types.nullOr types.path;
+        example = "/etc/php-fpm.conf";
+        description = ''
+          Specify alternative path to FastCGI process manager config file.
+        '';
+      };
+    };
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = [ pkg ];
+
+    systemd.services.php53-fpm = {
+      description = "php53-fpm service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        ExecStart = "${pkg}/sbin/php-fpm -F" +
+          valOpt cfg.prefix "prefix" +
+          valOpt cfg.configFile "fpm-config";
+        ExecReload = "${pkgs.coreutils}/bin/kill -USR2 $MAINPID";
+        Restart = "always";
+        Type = "simple";
+        KillMode = "process";
+      };
+    };
+  };
+
+}
+
+# vim: et ts=2 sw=2
+
