checkvpw-dovecot-wrapper

diff --git a/pkgs/tools/networking/vmailmgr/dovecot-wrapper.nix b/pkgs/tools/networking/vmailmgr/dovecot-wrapper.nix
new file mode 100644
index 0000000..d13dc77
--- /dev/null
+++ b/pkgs/tools/networking/vmailmgr/dovecot-wrapper.nix
@@ -0,0 +1,29 @@
+{ stdenv, fetchurl, vmailmgr }:
+
+let
+  pkg = "checkvpw-dovecot-wrapper";
+  ver = "0.1";
+  homepage = http://mij.oltrelinux.com/net/dovecot-qmail-vmailmgr;
+in stdenv.mkDerivation rec {
+  name = "${pkg}-${ver}";
+
+  src = fetchurl {
+    url = "${homepage}/checkvpw-dovecot-wrapper.c";
+    sha256 = "0s7i3nibzf64c1833zzhd3j7g8wnx5aqg1ifsl1kwqysfpyarclk";
+  };
+
+  buildCommand = ''
+    mkdir -p $out/bin
+    cp $src ./main.c
+    sed -i -e "s|/usr/local/bin/checkvpw|${vmailmgr}/bin/checkvpw|" ./main.c
+    $CC -Wall -O2 -o checkvpw-dovecot-wrapper ./main.c
+    install -s checkvpw-dovecot-wrapper $out/bin
+  '';
+
+  meta = {
+    description = "Integration of vmailmgr and dovecot";
+    homepage = homepage;
+    license = stdenv.lib.licenses.bsd2;
+  };
+}
+
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 0f5632f..935c27a 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3632,6 +3632,8 @@ let
 
   vmailmgr = callPackage ../tools/networking/vmailmgr {};
 
+  checkvpw-dovecot-wrapper = callPackage ../tools/networking/vmailmgr/dovecot-wrapper.nix {};
+
   vnc2flv = callPackage ../tools/video/vnc2flv {};
 
   vncrec = callPackage ../tools/video/vncrec { };
