Added tzeman to maintainer list

diff --git a/maintainers/maintainer-list.nix b/maintainers/maintainer-list.nix
index 181873314f3..59f0ed05bbf 100644
--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -3616,6 +3616,11 @@
     github = "twey";
     name = "James ‘Twey’ Kay";
   };
+  tzeman = {
+    email = "tzeman@volny.cz";
+    github = "tzeman77";
+    name = "Tomas Zeman";
+  };
   unode = {
     email = "alves.rjc@gmail.com";
     github = "unode";
