diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index d35c04e9c86..53cb6bb4742 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -8145,6 +8145,10 @@ in
     config.php.fpm = false;
   };
 
+  php73-bzip2 = php73.override {
+    config.php.bzip2 = true;
+  };
+
   inherit (callPackages ../development/interpreters/php/5.3.nix {})
     php53;
 
