From: Tomas Zeman <tzeman@volny.cz>
Subject: [PATCH] tg/bglibs

bglibs-1.106: One stop library package

diff --git a/pkgs/development/libraries/bglibs/bg-installer.patch b/pkgs/development/libraries/bglibs/bg-installer.patch
new file mode 100644
index 0000000..93718e4
--- /dev/null
+++ b/pkgs/development/libraries/bglibs/bg-installer.patch
@@ -0,0 +1,11 @@
+--- bglibs-1.106.orig/Makefile	2009-02-19 17:11:51.000000000 +0000
++++ bglibs-1.106/Makefile	2013-09-10 20:02:54.160098939 +0000
+@@ -90,7 +90,7 @@
+ 	./ltcompile base64/encode_part.c
+ 
+ bg-installer: bg-installer.o ltload libbg-cli.la libbg-msg.la libbg-path.la libbg-iobuf.la libbg-fmt.la libbg-str.la libbg-sysdeps.la
+-	./ltload bg-installer libbg-cli.la libbg-msg.la libbg-path.la libbg-iobuf.la libbg-fmt.la libbg-str.la libbg-sysdeps.la -static 
++	./ltload bg-installer cli/.libs/main.o libbg-cli.la libbg-msg.la libbg-path.la libbg-iobuf.la libbg-fmt.la libbg-str.la libbg-sysdeps.la -static 
+ 
+ bg-installer.o: compile bg-installer.c sysdeps.h cli/cli.h fmt/number.h iobuf/ibuf.h iobuf/common.h iobuf/obuf.h iobuf/iobuf.h msg/msg.h msg/debug.h msg/wrap.h path/path.h str/str.h
+ 	./compile bg-installer.c
diff --git a/pkgs/development/libraries/bglibs/bglibs.patch b/pkgs/development/libraries/bglibs/bglibs.patch
new file mode 100644
index 0000000..a0b4111
--- /dev/null
+++ b/pkgs/development/libraries/bglibs/bglibs.patch
@@ -0,0 +1,14 @@
+diff -ru bglibs-1.106.orig/Makefile bglibs-1.106/Makefile
+--- bglibs-1.106.orig/Makefile	2009-02-19 18:11:51.000000000 +0100
++++ bglibs-1.106/Makefile	2013-09-12 15:11:46.887246484 +0200
+@@ -145,8 +145,8 @@
+ 	) >compile
+ 	chmod 755 compile
+ 
+-crc-gentab: crc-gentab.o ltload libbg-msg.la libbg-iobuf.la libbg-sysdeps.la
+-	./ltload crc-gentab libbg-msg.la libbg-iobuf.la libbg-sysdeps.la -static 
++crc-gentab: crc-gentab.o ltload libbg-msg.la libbg-iobuf.la libbg-fmt.la libbg-str.la libbg-sysdeps.la
++	./ltload crc-gentab libbg-msg.la libbg-iobuf.la libbg-fmt.la libbg-str.la libbg-sysdeps.la -static 
+ 
+ crc-gentab.o: compile crc-gentab.c sysdeps.h iobuf/iobuf.h iobuf/common.h iobuf/ibuf.h iobuf/obuf.h msg/msg.h msg/debug.h
+ 	./compile crc-gentab.c
diff --git a/pkgs/development/libraries/bglibs/default.nix b/pkgs/development/libraries/bglibs/default.nix
new file mode 100644
index 0000000..5297117
--- /dev/null
+++ b/pkgs/development/libraries/bglibs/default.nix
@@ -0,0 +1,40 @@
+{ stdenv, fetchurl, libtool, which, perl }:
+
+let
+  pkg = "bglibs";
+  ver = "1.106";
+  homepage = http://untroubled.org/bglibs;
+in stdenv.mkDerivation rec {
+  name = "${pkg}-${ver}";
+
+  src = fetchurl {
+    url = "${homepage}/${name}.tar.gz";
+    sha256 = "1w4dagl2h1gps2kbzdk9zn3ls1c81q0d6cnczkr8zrc85lffb2jw";
+  };
+
+  buildInputs = [libtool which perl];
+
+  patches = [
+    ./bglibs.patch
+  ];
+
+  configurePhase = ''
+    echo "gcc -s -static" > conf-ld
+    echo $out/include/bglibs > conf-include
+    echo $out/lib > conf-lib
+    echo $out/bin > conf-bin
+    echo $out/man > conf-man
+  '';
+
+  meta = {
+    description = "One stop library package";
+    homepage = homepage;
+    license = stdenv.lib.licenses.lgpl21Plus.fullName;
+    platforms = stdenv.lib.platforms.gnu;
+    maintainers = with stdenv.lib.maintainers; [ tzeman ];
+  };
+
+}
+
+# vim: et ts=2 sw=2 
+
diff --git a/pkgs/development/libraries/bglibs/diet.nix b/pkgs/development/libraries/bglibs/diet.nix
new file mode 100644
index 0000000..6554768
--- /dev/null
+++ b/pkgs/development/libraries/bglibs/diet.nix
@@ -0,0 +1,45 @@
+{ stdenv, fetchurl, dietlibc, libtool, which, perl }:
+
+let
+  pkg = "bglibs";
+  ver = "1.106";
+  homepage = http://untroubled.org/bglibs;
+in stdenv.mkDerivation rec {
+  name = "${pkg}-${ver}";
+
+  src = fetchurl {
+    url = "${homepage}/${name}.tar.gz";
+    sha256 = "1w4dagl2h1gps2kbzdk9zn3ls1c81q0d6cnczkr8zrc85lffb2jw";
+  };
+
+  buildInputs = [dietlibc libtool which perl];
+
+  patches = [
+    # libtool diet gcc... can't find main.o inside .a library
+    ./bg-installer.patch
+  ];
+
+  configurePhase = ''
+    echo "diet gcc" > conf-cc
+    echo "diet gcc -s -static" > conf-ld
+    echo $out/include/bglibs > conf-include
+    echo $out/lib > conf-lib
+    echo $out/bin > conf-bin
+    echo $out/man > conf-man
+  '';
+
+  allowedReferences = ["out" perl];
+  dontStrip = true; # diet does not need stripping
+  dontPatchELF = true; # we produce static binaries
+
+  meta = {
+    description = "One stop library package";
+    homepage = homepage;
+    license = stdenv.lib.licenses.lgpl21Plus.fullName;
+    platforms = stdenv.lib.platforms.gnu;
+    maintainers = with stdenv.lib.maintainers; [ tzeman ];
+  };
+
+}
+
+# vim: et ts=2 sw=2 
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index cc942c3..2aa88db 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -513,6 +513,10 @@ let
 
   bfr = callPackage ../tools/misc/bfr { };
 
+  bglibs = callPackage ../development/libraries/bglibs { };
+
+  diet-bglibs = callPackage ../development/libraries/bglibs/diet.nix { };
+
   bmon = callPackage ../tools/misc/bmon { };
 
   boomerang = callPackage ../development/tools/boomerang {
