libowfat-0.29 (w/ dietlibc): reimplement libdjb under GPL

diff --git a/pkgs/development/libraries/libowfat/diet.nix b/pkgs/development/libraries/libowfat/diet.nix
new file mode 100644
index 0000000..cd33d45
--- /dev/null
+++ b/pkgs/development/libraries/libowfat/diet.nix
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, dietlibc }:
+
+let
+  branch = "diet";
+  pkg = "libowfat";
+  ver = "0.29";
+  homepage = http://www.fefe.de/libowfat;
+in stdenv.mkDerivation rec {
+  name = "${branch}-${pkg}-${ver}";
+
+  src = fetchurl {
+    url = "http://dl.fefe.de/${pkg}-${ver}.tar.bz2";
+    sha256 = "09v4phf1d4y617fdqwn214jmkialf7xqcsyx3rzk7x5ysvpbvbab";
+  };
+
+  buildInputs = [dietlibc];
+
+  installPhase = "make install prefix=$out LIBDIR=$out/lib";
+
+  allowedReferences = ["out"];
+  dontStrip = true; # diet does not need stripping
+  dontPatchELF = true; # we produce static binaries
+
+  meta = {
+    description = "reimplement libdjb under GPL";
+    longDescription = ''
+      libowfat is a library of general purpose APIs extracted from Dan
+      Bernstein's software, reimplemented and covered by the GNU General
+      Public License Version 2 (no later versions).
+    '';
+    homepage = homepage;
+    license = stdenv.lib.licenses.gpl2;
+    platforms = stdenv.lib.platforms.linux;
+    maintainers = with stdenv.lib.maintainers; [ tzeman ];
+  };
+
+}
+
+# vim: et ts=2 sw=2 
+
+
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index cbe67cd..b5ca7b3 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -10076,6 +10076,8 @@ let
 
   libowfat = callPackage ../development/libraries/libowfat { };
 
+  diet-libowfat = callPackage ../development/libraries/libowfat/diet.nix { };
+
   librecad = callPackage ../applications/misc/librecad { };
 
   librecad2 = callPackage ../applications/misc/librecad/2.0.nix { };
