diff --git a/modules/services/networking/dnscache.nix b/modules/services/networking/dnscache.nix
index f60c1ff..3e03591 100644
--- a/modules/services/networking/dnscache.nix
+++ b/modules/services/networking/dnscache.nix
@@ -82,6 +82,18 @@ in
 
     users.extraGroups.dnslog.gid = config.ids.gids.dnslog;
 
+    systemd.services.dnscache = {
+      description = "DNS cache service";
+      wantedBy = [ "multi-user.target" ];
+      environment = {
+        CACHESIZE = toString cfg.cacheSize;
+        DATALIMIT = toString cfg.dataLimit;
+        IP = cfg.listenIp;
+        IPSEND = cfg.sendIp;
+        ROOT = cfg.root;
+      };
+    };
+
   };
 
 }
