dnscache service

diff --git a/modules/misc/ids.nix b/modules/misc/ids.nix
index 5b545b7..2d04ce3 100644
--- a/modules/misc/ids.nix
+++ b/modules/misc/ids.nix
@@ -94,6 +94,9 @@ in
     bacula = 81;
     almir = 82;
     deluge = 83;
+    dnscache = 184;
+    tinydns = 185;
+    dnslog = 186;
 
     # When adding a uid, make sure it doesn't match an existing gid.
 
@@ -177,6 +180,9 @@ in
     # When adding a gid, make sure it doesn't match an existing uid.
 
     users = 100;
+    dnscache = 184;
+    tinydns = 185;
+    dnslog = 186;
     nixbld = 30000;
     nogroup = 65534;
   };
diff --git a/modules/module-list.nix b/modules/module-list.nix
index 09f53db..a260a40 100644
--- a/modules/module-list.nix
+++ b/modules/module-list.nix
@@ -145,6 +145,7 @@
   #./services/networking/dhclient.nix
   ./services/networking/dhcpcd.nix
   ./services/networking/dhcpd.nix
+  ./services/networking/dnscache.nix
   ./services/networking/dnsmasq.nix
   ./services/networking/ejabberd.nix
   ./services/networking/firewall.nix
diff --git a/modules/services/networking/dnscache.nix b/modules/services/networking/dnscache.nix
new file mode 100644
index 0000000..5383fae
--- /dev/null
+++ b/modules/services/networking/dnscache.nix
@@ -0,0 +1,57 @@
+{ config, pkgs, ... }:
+
+with pkgs.lib;
+
+let
+
+  cfg = config.services.dnscache;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.dnscache = {
+
+      enable = mkOption {
+        default = true;
+        description = "Whether to enable dnscache service.";
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.minit.enable = true;
+
+    environment.systemPackages = [ pkgs.daemontools pkgs.djbdns pkgs.minit ];
+
+    users.extraUsers.dnscache = {
+      uid = config.ids.uids.dnscache;
+      group = "dnscache";
+      description = "dnscache service user";
+    };
+
+    users.extraGroups.dnscache.gid = config.ids.gids.dnscache;
+
+    users.extraUsers.dnslog = {
+      uid = config.ids.uids.dnslog;
+      group = "dnslog";
+      description = "dnslog service user";
+    };
+
+    users.extraGroups.dnslog.gid = config.ids.gids.dnslog;
+  };
+
+}
+
+# vim: et ts=2 sw=2
+
