clamav: fix for respawning freshclam

diff --git a/nixos/modules/services/security/clamav.nix b/nixos/modules/services/security/clamav.nix
index 057891a..cecb620 100644
--- a/nixos/modules/services/security/clamav.nix
+++ b/nixos/modules/services/security/clamav.nix
@@ -55,26 +55,29 @@ in
       };
 
     services.clamav.updater.config = ''
-      DatabaseDirectory ${stateDir}
-      Foreground yes
-      Checks ${toString cfg.updater.frequency}
       DatabaseMirror database.clamav.net
     '';
 
-    jobs = {
-      clamav_updater = {
-	name = "clamav-updater";
-          startOn = "started network-interfaces";
-          stopOn = "stopping network-interfaces";
+    systemd.services.clamav_updater = {
+      description = "freshclam(1) service (clamav updater)";
+      wantedBy = [ "multi-user.target" ];
 
-          preStart = ''
-            mkdir -m 0755 -p ${stateDir}
-            chown ${clamavUser}:${clamavGroup} ${stateDir}
-          '';
-          exec = "${pkgs.clamav}/bin/freshclam --config-file=${pkgs.writeText "freshclam.conf" cfg.updater.config}";
-      }; 
-    };
+      preStart = ''
+        mkdir -m 0755 -p ${stateDir}
+        chown ${clamavUser}:${clamavGroup} ${stateDir}
+      '';
 
+      serviceConfig = {
+        ExecStart = ''
+          ${pkgs.clamav}/bin/freshclam -d --datadir=${stateDir} -c ${toString cfg.updater.frequency} \
+	    --config-file=${pkgs.writeText "freshclam.conf" cfg.updater.config}
+        '';
+        Restart = "always";
+        Type = "forking";
+      };
+    };
   };
 
-}
\ No newline at end of file
+}
+
+# vim: et ts=2 sw=2 
