Added tzeman to maintainer list

diff --git a/lib/maintainers.nix b/lib/maintainers.nix
index 5061cab..6159904 100644
--- a/lib/maintainers.nix
+++ b/lib/maintainers.nix
@@ -75,6 +75,7 @@
   thoughtpolice = "Austin Seipp <aseipp@pobox.com>";
   tomberek = "Thomas Bereknyei <tomberek@gmail.com>";
   ttuegel = "Thomas Tuegel <ttuegel@gmail.com>";
+  tzeman = "Tomáš Zeman <tzeman@volny.cz>";
   urkud = "Yury G. Kudryashov <urkud+nix@ya.ru>";
   vcunat = "Vladimír Čunát <vcunat@gmail.com>";
   viric = "Lluís Batlle i Rossell <viric@viric.name>";
