Added tzeman to maintainer list

diff --git a/lib/maintainers.nix b/lib/maintainers.nix
index 82986fa..57db993 100644
--- a/lib/maintainers.nix
+++ b/lib/maintainers.nix
@@ -162,6 +162,7 @@
   ttuegel = "Thomas Tuegel <ttuegel@gmail.com>";
   tv = "Tomislav Viljetić <tv@shackspace.de>";
   twey = "James ‘Twey’ Kay <twey@twey.co.uk>";
+  tzeman = "Tomáš Zeman <tzeman@volny.cz>";
   urkud = "Yury G. Kudryashov <urkud+nix@ya.ru>";
   vandenoever = "Jos van den Oever <jos@vandenoever.info>";
   vbgl = "Vincent Laporte <Vincent.Laporte@gmail.com>";
