Added tzeman to maintainer list

diff --git a/lib/maintainers.nix b/lib/maintainers.nix
index 0418337..beff65c 100644
--- a/lib/maintainers.nix
+++ b/lib/maintainers.nix
@@ -56,6 +56,7 @@
   thammers = "Tobias Hammerschmidt <jawr@gmx.de>";
   the-kenny = "Moritz Ulrich <moritz@tarn-vedra.de>";
   tomberek = "Thomas Bereknyei <tomberek@gmail.com>";
+  tzeman = "Tomáš Zeman <tzeman@volny.cz>";
   urkud = "Yury G. Kudryashov <urkud+nix@ya.ru>";
   vcunat = "Vladimír Čunát <vcunat@gmail.com>";
   viric = "Lluís Batlle i Rossell <viric@viric.name>";
