minit module/service unit

diff --git a/modules/module-list.nix b/modules/module-list.nix
index 20ad0a0..09f53db 100644
--- a/modules/module-list.nix
+++ b/modules/module-list.nix
@@ -196,6 +196,7 @@
   ./services/security/torsocks.nix
   ./services/system/dbus.nix
   ./services/system/kerberos.nix
+  ./services/system/minit.nix
   ./services/system/nscd.nix
   ./services/system/uptimed.nix
   ./services/torrent/deluge.nix
diff --git a/modules/services/system/minit.nix b/modules/services/system/minit.nix
new file mode 100644
index 0000000..70d304e
--- /dev/null
+++ b/modules/services/system/minit.nix
@@ -0,0 +1,60 @@
+{ config, pkgs, ... }:
+
+with pkgs.lib;
+
+let
+
+  cfg = config.services.minit;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.minit = {
+
+      enable = mkOption {
+        default = true;
+        description = "Whether to enable minit service.";
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf config.services.minit.enable {
+
+    environment.systemPackages = [ pkgs.minit ];
+
+    systemd.services.minit =
+      { description = "Minit service";
+
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+
+        preStart =
+          ''
+            dir=/etc/minit
+            mkdir -p $dir
+            for i in 'in' 'out'; do
+              [ -p $dir/$i ] || mkfifo -m 600 $dir/$i
+            done
+          '';
+
+        serviceConfig = {
+          Type = "simple";
+          ExecStart = "${pkgs.minit}/sbin/minit";
+        };
+      };
+
+  };
+
+}
+
+# vim: et ts=2 sw=2
