minit: fix PATH

diff --git a/nixos/modules/services/system/minit.nix b/nixos/modules/services/system/minit.nix
index f84d426..6215f0b 100644
--- a/nixos/modules/services/system/minit.nix
+++ b/nixos/modules/services/system/minit.nix
@@ -34,6 +34,9 @@ in
 
     systemd.services.minit =
       { description = "Minit service";
+        environment = {
+          PATH = "/run/current-system/sw/bin:/run/current-system/sw/sbin";
+        };
 
         wantedBy = [ "multi-user.target" ];
         after = [ "network.target" ];
