nginx exec script wrapped

diff --git a/nixos/modules/services/web-servers/nginx/default.nix b/nixos/modules/services/web-servers/nginx/default.nix
index 1a39fe4..ba5b00c 100644
--- a/nixos/modules/services/web-servers/nginx/default.nix
+++ b/nixos/modules/services/web-servers/nginx/default.nix
@@ -75,6 +75,14 @@ in
   config = mkIf cfg.enable {
     environment.systemPackages = [ nginx ];
 
+    environment.etc."nginx_wrapper" = {
+      text = ''
+        #!/bin/sh
+        ${nginx}/bin/nginx -c ${configFile} -p ${cfg.stateDir} $@
+      '';
+      mode = "0755";
+    };
+
     # TODO: test user supplied config file pases syntax test
 
     systemd.services.nginx = {
