php-apc for php_fpm53

diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 040769a..392de7e 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -4107,6 +4107,10 @@ let
 
   php_apc = callPackage ../development/libraries/php-apc { };
 
+  php_fpm53_apc = callPackage ../development/libraries/php-apc {
+    php = php_fpm53;
+  };
+
   picolisp = callPackage ../development/interpreters/picolisp {};
 
   pltScheme = racket; # just to be sure
