diff --git a/pkgs/development/interpreters/php/5.3.nix b/pkgs/development/interpreters/php/5.3.nix
index 3f38778..3b5e6d4 100644
--- a/pkgs/development/interpreters/php/5.3.nix
+++ b/pkgs/development/interpreters/php/5.3.nix
@@ -233,11 +233,7 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
 #       configureFlags = ["--with-tidy=shared"];
 #     };
 
-      ttf = {
-        configureFlags = ["--with-ttf=shared"];
-      };
-
-      xmlrpc = {
+      mlrpc = {
         configureFlags = ["--with-xmlrpc=shared"];
       };
     };
@@ -279,7 +275,6 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
     socketsSupport = config.php.sockets or true;
     sqliteSupport = config.php.sqlite or true;
 #   tidySupport = config.php.tidy or true;
-    ttfSupport = config.php.ttf or true;
     xmlrpcSupport = config.php.xmlrpc or true;
     xslSupport = config.php.xsl or true;
     zipSupport = config.php.zip or true;
@@ -291,7 +286,7 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
     [[ -z "$libxml2" ]] || export PATH=$PATH:$libxml2/bin
     ./configure --with-config-file-scan-dir=/etc --with-config-file-path=$out/etc \
         --prefix=$out $configureFlags \
-        --enable-shared --enable-magic-quotes --enable-safe-mode --with-mime-magic
+        --enable-shared --enable-magic-quotes --enable-safe-mode
     echo configurePhase end
   '' + stdenv.lib.optionalString stdenv.isDarwin ''
     # don't build php.dSYM as the php binary
