php53: depend on mysql55 instead of mysql

diff --git a/pkgs/development/interpreters/php/5.3.nix b/pkgs/development/interpreters/php/5.3.nix
index c4da63e..3f38778 100644
--- a/pkgs/development/interpreters/php/5.3.nix
+++ b/pkgs/development/interpreters/php/5.3.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchurl, composableDerivation, autoconf, automake, flex, bison
-, apacheHttpd, mysql, libxml2, readline, zlib, curl, gd, postgresql, gettext
+, apacheHttpd, mysql55, libxml2, readline, zlib, curl, gd, postgresql, gettext
 , openssl, pkgconfig, sqlite, config, libjpeg, libpng, freetype, libxslt
 , libmcrypt, bzip2, icu, libssh2, makeWrapper, libiconvOrEmpty, libiconv, uwimap
 , gdbm, gmp
@@ -99,15 +99,15 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
 
       mysql = {
         configureFlags = [
-          "--with-mysql=shared,${mysql}"
+          "--with-mysql=shared,${mysql55}"
           "--with-mysql-sock=/tmp/mysql.sock"
         ];
-        buildInputs = [ mysql ];
+        buildInputs = [ mysql55 ];
       };
 
       mysqli = {
-        configureFlags = ["--with-mysqli=${mysql}/bin/mysql_config"];
-        buildInputs = [ mysql];
+        configureFlags = ["--with-mysqli=${mysql55}/bin/mysql_config"];
+        buildInputs = [ mysql55];
       };
 
       mysqli_embedded = {
@@ -117,8 +117,8 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
       };
 
       pdo_mysql = {
-        configureFlags = ["--with-pdo-mysql=shared,${mysql}"];
-        buildInputs = [ mysql ];
+        configureFlags = ["--with-pdo-mysql=shared,${mysql55}"];
+        buildInputs = [ mysql55 ];
       };
 
       bcmath = {
