PHP 5.3: fix url

diff --git a/pkgs/development/interpreters/php/5.3.nix b/pkgs/development/interpreters/php/5.3.nix
index 58e9816..ad653b4 100644
--- a/pkgs/development/interpreters/php/5.3.nix
+++ b/pkgs/development/interpreters/php/5.3.nix
@@ -302,7 +302,7 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
   '' );
 
   src = fetchurl {
-    url = "http://www.php.net/distributions/php-${version}.tar.bz2";
+    url = "http://cz1.php.net/get/php-${version}.tar.bz2/from/this/mirror";
     sha256 = "1480pfp4391byqzmvdmbxkdkqwdzhdylj63sfzrcgadjf9lwzqf4";
     name = "php-${version}.tar.bz2";
   };
