php53: mcrypt enable

diff --git a/pkgs/development/interpreters/php/5.3.nix b/pkgs/development/interpreters/php/5.3.nix
index cd220f5..c4da63e 100644
--- a/pkgs/development/interpreters/php/5.3.nix
+++ b/pkgs/development/interpreters/php/5.3.nix
@@ -259,7 +259,7 @@ composableDerivation.composableDerivation {} ( fixed : let inherit (fixed.fixed)
     intlSupport = config.php.intl or true;
     libxml2Support = config.php.libxml2 or true;
     mbstringSupport = config.php.mbstring or true;
-    mcryptSupport = config.php.mcrypt or false;
+    mcryptSupport = config.php.mcrypt or true;
     mhashSupport = config.php.mhash or false;
     mysqlSupport = config.php.mysql or true;
     mysqliSupport = config.php.mysqli or true;
