From: Tomas Zeman <tzeman@volny.cz>
Subject: [PATCH] tg/ucspi-tcp

ucspi-tcp-0.88: compiled w/ dietlibc

diff --git a/pkgs/tools/networking/ucspi-tcp/diet-compile.patch b/pkgs/tools/networking/ucspi-tcp/diet-compile.patch
new file mode 100644
index 0000000..f2bbb2c
--- /dev/null
+++ b/pkgs/tools/networking/ucspi-tcp/diet-compile.patch
@@ -0,0 +1,13 @@
+diff -ru ucspi-tcp-0.88.orig/Makefile ucspi-tcp-0.88/Makefile
+--- ucspi-tcp-0.88.orig/Makefile	2000-03-18 16:18:42.000000000 +0100
++++ ucspi-tcp-0.88/Makefile	2013-09-30 12:49:28.847991546 +0200
+@@ -328,8 +328,7 @@
+ hasshsgr.h: \
+ choose compile load tryshsgr.c hasshsgr.h1 hasshsgr.h2 chkshsgr \
+ warn-shsgr
+-	./chkshsgr || ( cat warn-shsgr; exit 1 )
+-	./choose clr tryshsgr hasshsgr.h1 hasshsgr.h2 > hasshsgr.h
++	cat hasshsgr.h1 > hasshsgr.h
+ 
+ haswaitp.h: \
+ choose compile load trywaitp.c haswaitp.h1 haswaitp.h2
diff --git a/pkgs/tools/networking/ucspi-tcp/diet.nix b/pkgs/tools/networking/ucspi-tcp/diet.nix
new file mode 100644
index 0000000..e1036ed
--- /dev/null
+++ b/pkgs/tools/networking/ucspi-tcp/diet.nix
@@ -0,0 +1,64 @@
+{ stdenv, fetchurl, dietlibc }:
+
+let
+  brand = "diet";
+  pkg = "ucspi-tcp";
+  ver = "0.88";
+in stdenv.mkDerivation rec {
+  name = "${brand}-${pkg}-${ver}";
+
+  srcs = [
+    (fetchurl {
+      url = "http://cr.yp.to/ucspi-tcp/${pkg}-${ver}.tar.gz";
+      sha256 = "171yl9kfm8w7l17dfxild99mbf877a9k5zg8yysgb1j8nz51a1ja";
+    })
+    (fetchurl {
+      url = "http://smarden.org/pape/djb/manpages/${pkg}-${ver}-man.tar.gz";
+      sha256 = "1vq8nibyjq8cirif8iv2bpzp5jv0jpgjgfqa0cnbc8ipr9lxvc89";
+    })
+  ];
+  sourceRoot = "${pkg}-${ver}";
+
+  ipv6patch = fetchurl {
+    url = "http://www.fefe.de/ucspi/${pkg}-${ver}-ipv6.diff19.bz2";
+    sha256 = "1kls3xysv3wwmikxychnbfv8vbnb0ih15005hhn4a56pj392r59m";
+  };
+
+  patches = [
+    # Avoid need to have secondary groups for compilation
+    ./diet-compile.patch
+    # IPv6 support
+    ipv6patch
+  ];
+
+  buildInputs = [dietlibc];
+
+  configurePhase = ''
+    echo $out > conf-home
+    echo "diet gcc" > conf-cc
+    echo "diet gcc -s -static" > conf-ld
+  '';
+
+  installTargets = "setup check";
+
+  allowedReferences = ["out"];
+  dontPatchShebangs = true; # /bin/sh
+  dontStrip = true; # diet does not need stripping
+  dontPatchELF = true; # we produce static binaries
+
+  # man pages
+  postInstall = ''
+    mkdir -p $out/man/man1
+    cp $NIX_BUILD_TOP/${pkg}-${ver}-man/*.1 $out/man/man1/
+  '';
+
+  meta = {
+    description = "Command-line tools for building TCP client-server applications";
+    homepage = http://cr.yp.to/ucspi-tcp.html;
+    license = stdenv.lib.licenses.publicDomain.shortName;
+    platforms = stdenv.lib.platforms.gnu;
+    maintainers = with stdenv.lib.maintainers; [ tzeman ];
+  };
+}
+
+# vim: et ts=2 sw=2 
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index a76408d..ba3e5c6 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2575,6 +2575,8 @@ let
 
   ucspi-tcp = callPackage ../tools/networking/ucspi-tcp { };
 
+  diet-ucspi-tcp = callPackage ../tools/networking/ucspi-tcp/diet.nix { };
+
   udftools = callPackage ../tools/filesystems/udftools {};
 
   udptunnel = callPackage ../tools/networking/udptunnel { };
