mysql suport for zabbix 1.8

diff --git a/nixos/modules/services/monitoring/zabbix-server.nix b/nixos/modules/services/monitoring/zabbix-server.nix
index ca283ea..48cdaa3 100644
--- a/nixos/modules/services/monitoring/zabbix-server.nix
+++ b/nixos/modules/services/monitoring/zabbix-server.nix
@@ -34,7 +34,10 @@ let
       ''}
     '';
 
-  useLocalPostgres = cfg.dbServer == "localhost" || cfg.dbServer == "";
+  zbx = cfg.package;
+
+  useLocalPostgres = !cfg.mysql && (cfg.dbServer == "localhost" || cfg.dbServer == "");
+  useLocalMysql = cfg.mysql && (cfg.dbServer == "localhost" || cfg.dbServer == "");
 
 in
 
@@ -64,6 +67,17 @@ in
       description = "Password used to connect to the database server.";
     };
 
+    services.zabbixServer.mysql = mkOption {
+      default = false;
+      description = "User MySQL backend";
+    };
+
+    services.zabbixServer.package = mkOption {
+      type = types.package;
+      default = pkgs.zabbix.server;
+      description = "Zabbix package to use";
+    };
+
   };
 
   ###### implementation
@@ -71,6 +85,7 @@ in
   config = mkIf cfg.enable {
 
     services.postgresql.enable = useLocalPostgres;
+    services.mysql.enable = useLocalMysql;
 
     users.extraUsers = singleton
       { name = "zabbix";
@@ -89,19 +104,21 @@ in
             mkdir -m 0755 -p ${stateDir} ${logDir} ${libDir}
             chown zabbix ${stateDir} ${logDir} ${libDir}
 
-            if ! test -e "${libDir}/db-created"; then
+	    exit
+
+            if ! test -e "${libDir}/db-created" ; then
                 ${pkgs.postgresql}/bin/createuser --no-superuser --no-createdb --no-createrole zabbix || true
                 ${pkgs.postgresql}/bin/createdb --owner zabbix zabbix || true
-                cat ${pkgs.zabbix.server}/share/zabbix/db/schema/postgresql.sql | ${pkgs.su}/bin/su -s "$SHELL" zabbix -c '${pkgs.postgresql}/bin/psql zabbix'
-                cat ${pkgs.zabbix.server}/share/zabbix/db/data/images_pgsql.sql | ${pkgs.su}/bin/su -s "$SHELL" zabbix -c '${pkgs.postgresql}/bin/psql zabbix'
-                cat ${pkgs.zabbix.server}/share/zabbix/db/data/data.sql | ${pkgs.su}/bin/su -s "$SHELL" zabbix -c '${pkgs.postgresql}/bin/psql zabbix'
+                cat ${zbx}/share/zabbix/db/schema/postgresql.sql | ${pkgs.su}/bin/su -s "$SHELL" zabbix -c '${pkgs.postgresql}/bin/psql zabbix'
+                cat ${zbx}/share/zabbix/db/data/images_pgsql.sql | ${pkgs.su}/bin/su -s "$SHELL" zabbix -c '${pkgs.postgresql}/bin/psql zabbix'
+                cat ${zbx}/share/zabbix/db/data/data.sql | ${pkgs.su}/bin/su -s "$SHELL" zabbix -c '${pkgs.postgresql}/bin/psql zabbix'
                 touch "${libDir}/db-created"
             fi
           '';
 
         path = [ pkgs.nettools ];
 
-        serviceConfig.ExecStart = "@${pkgs.zabbix.server}/sbin/zabbix_server zabbix_server --config ${configFile}";
+        serviceConfig.ExecStart = "@${zbx}/sbin/zabbix_server zabbix_server --config ${configFile}";
         serviceConfig.Type = "forking";
         serviceConfig.Restart = "always";
         serviceConfig.RestartSec = 2;
diff --git a/pkgs/servers/monitoring/zabbix/default.nix b/pkgs/servers/monitoring/zabbix/default.nix
index 1e46ffd..3963a2f 100644
--- a/pkgs/servers/monitoring/zabbix/default.nix
+++ b/pkgs/servers/monitoring/zabbix/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, pkgconfig, postgresql, curl, openssl, zlib }:
+{ stdenv, fetchurl, pkgconfig, postgresql, curl, openssl, zlib, mysql ? null }:
 
 let
 
@@ -18,6 +18,10 @@ let
         ''}
     '';
 
+  withDb = if mysql == null then "--with-pgsql" else "--with-mysql";
+
+  inputDb = if mysql == null then postgresql else mysql;
+
 in
 
 {
@@ -27,9 +31,14 @@ in
 
     inherit src preConfigure;
 
-    configureFlags = "--enable-agent --enable-server --with-pgsql --with-libcurl";
+    configureFlags = [
+      "--enable-agent"
+      "--enable-server"
+      "--with-libcurl"
+      withDb
+    ];
 
-    buildInputs = [ pkgconfig postgresql curl openssl zlib ];
+    buildInputs = [ pkgconfig curl openssl zlib inputDb ];
 
     postInstall =
       ''
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index c015328..43903e3 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -6805,6 +6805,10 @@ let
     inherit fetchurl stdenv pkgconfig postgresql curl openssl zlib;
   });
 
+  zabbix18_mysql = callPackage ../servers/monitoring/zabbix {
+    mysql = mysql55;
+  };
+
   zabbix20 = callPackage ../servers/monitoring/zabbix/2.0.nix { };
   zabbix22 = callPackage ../servers/monitoring/zabbix/2.2.nix { };
 
