pywhois: Python whois library

diff --git a/pkgs/development/python-modules/pywhois/default.nix b/pkgs/development/python-modules/pywhois/default.nix
new file mode 100644
index 00000000000..2741aa434aa
--- /dev/null
+++ b/pkgs/development/python-modules/pywhois/default.nix
@@ -0,0 +1,22 @@
+{ stdenv, fetchurl, buildPythonPackage, future }:
+
+buildPythonPackage rec {
+  pname = "pywhois"; # -${version}";
+  version = "2019-09-07";
+
+  doCheck = false; # fails to pass because it tries to run in home directory
+
+  src = fetchurl {
+    url = "https://github.com/richardpenman/pywhois/archive/c6b441a3e9ae7d66d34166ba801009f00f72291c.zip";
+    sha256 = "0yr7yp0hzdysaxrzky2f5nv0kngjxw4i5ynkmbygvhj6jh9skd0q";
+  };
+
+  propagatedBuildInputs = [ future ];
+
+  meta = with stdenv.lib; {
+    description = "Python WHOIS library";
+    homepage = https://github.com/richardpenman/pywhois;
+  };
+}
+
+# vim: et ts=2 sw=2 
diff --git a/pkgs/top-level/python-packages.nix b/pkgs/top-level/python-packages.nix
index 5dc17f20f28..19a44512ae9 100644
--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -6258,6 +6258,9 @@ in {
   runway-python = callPackage ../development/python-modules/runway-python { };
 
   pyprof2calltree = callPackage ../development/python-modules/pyprof2calltree { };
+
+  pywhois = callPackage ../development/python-modules/pywhois { };
+
 });
 
 in fix' (extends overrides packages)
