nginx exec script wrapped

diff --git a/nixos/modules/services/web-servers/nginx/default.nix b/nixos/modules/services/web-servers/nginx/default.nix
index 7c2d3a4..1a01d5e 100644
--- a/nixos/modules/services/web-servers/nginx/default.nix
+++ b/nixos/modules/services/web-servers/nginx/default.nix
@@ -84,6 +84,14 @@ in
   };
 
   config = mkIf cfg.enable {
+    environment.etc."nginx_wrapper" = {
+      text = ''
+        #!/bin/sh
+        ${nginx}/bin/nginx -c ${configFile} -p ${cfg.stateDir} $@
+      '';
+      mode = "0755";
+    };
+
     # TODO: test user supplied config file pases syntax test
 
     systemd.services.nginx = {
