qmail logging

diff --git a/nixos/modules/services/mail/qmail.nix b/nixos/modules/services/mail/qmail.nix
index e2ed1ee..dd39d8e 100644
--- a/nixos/modules/services/mail/qmail.nix
+++ b/nixos/modules/services/mail/qmail.nix
@@ -103,6 +103,21 @@ in
         '';
       };
 
+      logDir = mkOption {
+        default = "/var/log/qmail/send";
+        description = "qmail log dir";
+      };
+
+      logSize = mkOption {
+        default = 1048576;
+        description = "qmail log file size, see `s' parameter of multilog(8)";
+      };
+
+      logNum = mkOption {
+        default = 10;
+        description = "qmail log file number, see `n' parameter of multilog(8)";
+      };
+
     };
   };
 
@@ -176,21 +191,54 @@ in
           n=`basename $i`
           [ -e $bdir/$n ] || ln -s /run/current-system/sw/bin/$n $bdir/$n
         done
+        if [ ! -p /run/.qmailsend ]; then
+          mkfifo -m600 /run/.qmailsend
+          chown qmaill:nofiles /run/.qmailsend
+        fi
       '';
 
       serviceConfig = {
-        ExecStart = "${pkg}/bin/qmail-start ${cfg.defaultdelivery}";
+        ExecStart = 
+          "${pkgs.bash}/bin/sh -c '" +
+          "exec 1> /run/.qmailsend; " +
+          "exec 2>&1; " +
+          "exec ${pkg}/bin/qmail-start ${cfg.defaultdelivery}" +
+          "'";
         Restart = "always";
         Type = "simple";
         KillMode = "process";
       };
     };
 
+    systemd.services.qmail-log = {
+      description = "qmail logging service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      preStart = ''
+        mkdir -p ${cfg.logDir}
+        chown -R qmaill:nofiles ${cfg.logDir}
+      '';
+
+      serviceConfig = {
+        ExecStart =
+          "${pkgs.bash}/bin/sh -c 'exec </run/.qmailsend; " +
+          "exec ${dtools}/bin/setuidgid qmaill " +
+          "${dtools}/bin/multilog t n${toString cfg.logNum} s${toString cfg.logSize} ${cfg.logDir} " +
+          "'";
+        Restart = "always";
+        Type = "simple";
+      };
+    };
+
     systemd.services.smtp-local = {
       description = "smtp local service";
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
       path = [ pkg ];
+      environment = {
+        RELAYCLIENT = "";
+      };
 
       serviceConfig = {
         ExecStart = "${dtools}/bin/softlimit -m 20000000 " +
